extensions/redis-cache/src/main/java/io/druid/client/cache/RedisCache.java

```diff
@@ -0,0 +1,288 @@
+        timeoutCount.incrementAndGet();
+      }
+
+      throw Throwables.propagate(ex);
+    }
+    catch (JedisException ex) {
+      errorCount.incrementAndGet();
+      log.warn(ex, "Unable to get cache items");
+      return results;
+    }
+
+    final int length = valuesList.size();
+    int i = 0;
+
+    while (i < length) {
+      final byte[] value = valuesList.get(i);
+      if (value != null) {
+        final NamedKey key = keyLookup.get(keysList.get(i));
+        try {
+          final byte[] deserialized = deserialize(key, value);
+          hitCount.incrementAndGet();
+          results.put(key, deserialized);
+        }
+        catch (LZ4Exception ex) {
+          errorCount.incrementAndGet();
+          log.warn(ex, "Error decompressing cache item");
+        }
+      }
+      else {
+        missCount.incrementAndGet();
+      }
+
+      i++;
+    }
+
+    return results;
+  }
+
+  @Override
+  public void close(String namespace)
+  {
+    final byte[] pattern = redisKey(prefix, hash(namespace), "*");
+    try (final Jedis client = pool.getResource()) {
+      final Set<byte[]> keys = client.keys(pattern);
+      if (keys.size() == 0) {
+        return;
+      }
+
+      client.del(keys.toArray(new byte[keys.size()][]));
+    }
+    catch (JedisConnectionException ex) {
+      Throwables.propagate(ex);
+    }
+    catch (JedisException ex) {
+      log.warn(ex, "Error cleaning cache items");
+    }
+  }
+
+  @Override
+  public CacheStats getStats()
+  {
+    return new CacheStats(
+        hitCount.get(),
+        missCount.get(),
+        0,
+        0,
+        0,
+        timeoutCount.get(),
+        errorCount.get()
+    );
+  }
+
+  @Override
+  public boolean isLocal()
+  {
+    return false;
+  }
+
+  @Override
+  public void doMonitor(ServiceEmitter emitter)
+  {
+    //TODO
+  }
+
+  private static boolean isTimeout(JedisConnectionException ex)
+  {
+    //FIXME NoSuchElementException is thrown not only in case of timeout
+    return ex.getCause() instanceof NoSuchElementException;
+  }
+
+  private byte[] deserialize(NamedKey key, byte[] bytes)
+  {
+    final byte[] data = transcoder.decompress(bytes);
+    final ByteBuffer buf = ByteBuffer.wrap(data);
+
+    final int keyLength = buf.getInt();
+    final byte[] keyBytes = new byte[keyLength];
+    buf.get(keyBytes);
+
+    final byte[] value = new byte[buf.remaining()];
+    buf.get(value);
+
+    Preconditions.checkState(Arrays.equals(keyBytes, key.toByteArray()),
+      "Keys do not match, possible hash collision?");
+
+    return value;
+  }
+
+  private byte[] serialize(NamedKey key, byte[] value)
+  {
+    final byte[] keyBytes = key.toByteArray();
+    final byte[] data = ByteBuffer.allocate(Ints.BYTES + keyBytes.length + value.length)
+      .putInt(keyBytes.length)
+      .put(keyBytes)
+      .put(value)
+      .array();
+
+    return transcoder.compress(data);
+  }
+
+  private static byte[] computeKeyHash(String prefix, NamedKey key)
```

Comment:
it might be useful to share the code with memcached cache here instead of having two different implementation, unless there is a good reason to have two different ones.